### PR TITLE
fixes - ast review by Alexey

### PR DESCRIPTION
**Pull request by olgakil**
minor fixes - code review AST branch by Alexey

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1010625&projectid=206)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|112
Low|1415
Informational|38

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

File: src/main/java/com/checkmarx/flow/CxFlowRunner.java [hunk 1]
```diff
@@ -484,11 +484,13 @@ private ScanResults scan(ScanRequest request) throws ExitThrowable {
     }
 
     private boolean isScaEnabled() {
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/CxFlowRunner.java:486]: each VulnerabilityScanner has a isEnabled function.
having this function in CxFlowRunner is a logical bug!

File: src/main/java/com/checkmarx/flow/CxFlowRunner.java [hunk 2]
```diff
@@ -484,11 +484,13 @@ private ScanResults scan(ScanRequest request) throws ExitThrowable {
...
-        return flowProperties.getEnabledVulnerabilityScanners().contains(SCA_SCANNER);
+        return (flowProperties.getEnabledVulnerabilityScanners() != null)
+          && flowProperties.getEnabledVulnerabilityScanners().contains(SCA_SCANNER);
     }
 
     private boolean isAstEnabled() {
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/CxFlowRunner.java:491]: each VulnerabilityScanner has a isEnabled function.
having this function in CxFlowRunner is a logical bug!

File: src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java [hunk 3]
```diff
@@ -53,15 +53,19 @@ public ScanResults scan(ScanRequest scanRequest) {
...
-            log.error(message, e);
-            OperationResult scanCreationFailure = new OperationResult(OperationStatus.FAILURE, e.getMessage());
-            logRequest(scanRequest, getScanId(internalResults),  scanCreationFailure);
-            throw new MachinaRuntimeException(message);
+            return treatError(scanRequest, internalResults, e);
         }
         return result;
     }
 
+    private ScanResults treatError(ScanRequest scanRequest, ASTResultsWrapper internalResults, Exception e) {
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java:61]: This function never returns ScanResults

File: src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java [hunk 4]
```diff
@@ -53,15 +53,19 @@ public ScanResults scan(ScanRequest scanRequest) {
             logRequest(scanRequest, getScanId(internalResults),  OperationResult.successful());
             result = toScanResults(internalResults);
         } catch (Exception e) {
-            final String message = "SCA scan failed.";
-            log.error(message, e);
-            OperationResult scanCreationFailure = new OperationResult(OperationStatus.FAILURE, e.getMessage());
-            logRequest(scanRequest, getScanId(internalResults),  scanCreationFailure);
-            throw new MachinaRuntimeException(message);
+            return treatError(scanRequest, internalResults, e);
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java:56]: please don't return here!!
also this function never returns

File: src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java [hunk 5]
```diff
@@ -85,11 +89,7 @@ public ScanResults scan(ScanRequest scanRequest, String path) throws ExitThrowab
             Files.deleteIfExists(Paths.get(cxZipFile));
 
         } catch (Exception e) {
-            final String message = "SCA scan failed.";
-            log.error(message, e);
-            OperationResult scanCreationFailure = new OperationResult(OperationStatus.FAILURE, e.getMessage());
-            logRequest(scanRequest, getScanId(internalResults),  scanCreationFailure);
-            throw new MachinaRuntimeException(message);
+            return treatError(scanRequest, internalResults, e);
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java:92]: same error as in the other throw

**Pull request merged**